### DISCUSSION TITLE
Feature proposal: Find (in files) history

### DISCUSSION THREAD

**Discussion by GhoulFool** [godotengine/godot-proposals#7570] [Editor]
I find the the find and find in all files work well, but could benefit from retaining a history. That is the last 5 or 6 recent searches which could be accessed by an arrow in the search dialog box.
Currently:
![image](https://github.com/godotengine/godot-proposals/assets/25694861/f5a5def1-7bf5-458e-ad48-35a28cd22944)
with history
![image](https://github.com/godotengine/godot-proposals/assets/25694861/8e26fb40-e5fc-4787-a8d5-99e5d1c3ffe4)

**Comment by Calinou** [comment]
I'd do it like VS Code where pressing up/down arrows while on the text field brings up the previous/next search terms. This can be extended to all search fields in the editor, but it'll be the most useful in the script editor's find dialog as well as Find in Files (+ Replace in Files).

History will need to be saved to disk (likely in the cache folder) so that it can be recalled across editor restarts.
